packages/teuchos/numerics/src/Teuchos_SerialBandDenseMatrix.hpp

```diff
@@ -1075,6 +1070,16 @@ void SerialBandDenseMatrix<OrdinalType, ScalarType>::copyMat(
   }
 }
 
+#ifndef TEUCHOS_HIDE_DEPRECATED_CODE
```

Comment:
For consistency with Kokkos, should we instead call the macro `TEUCHOS_ENABLE_DEPRECATED_CODE`, do an `#ifdef` test, and have the macro be defined by default?

Reply:
@mhoemmen, actually, this is consistent with the original standard documented at:

* https://tribits.org/doc/TribitsDevelopersGuide.html#hiding-deprecated-code-to-certify-and-facilitate-later-removal

By requiring a user to define `TEUCHOS_ENABLE_DEPRECATED_CODE` you are essentially breaking backward compatibility.  (Of course the configured header file should set this define correctly but requiring a macro to be defined to hide deprecated seems to better avoid breaks in backward compatibility.)

Reply:
@bartlettroscoe Hm, Kokkos has already chosen the `ENABLE` path; maybe we should work something out with Christian so that it's consistent across packages.

Reply:
@mhoemmen, I don't think it really matters what macro names are used as long as the package configures its `<Package>_config.h` file to define its macros correctly according to the cmake configure options defined at:

* https://tribits.org/doc/TribitsBuildReference.html#disabling-deprecated-code

We could look in the git repo histories but I am pretty sure the stuff documented at:

* https://tribits.org/doc/TribitsDevelopersGuide.html#regulated-backward-compatibility-and-deprecated-code

existed before Kokkos added deprecated support.